test/methods.jl

```diff
@@ -1,70 +1,46 @@
-import FiniteDifferences: estimate_magitude, estimate_roundoff_error
-
 @testset "Methods" begin
-    @testset "estimate_magitude" begin
-        f64(x::Float64) = x
-        f64_int(x::Float64) = Int(10x)
-        @test estimate_magitude(f64, 0.0) === 0.1
-        @test estimate_magitude(f64, 1.0) === 1.0
-        @test estimate_magitude(f64_int, 0.0) === 1.0
-        @test estimate_magitude(f64_int, 1.0) === 10.0
-
-        f32(x::Float32) = x
-        f32_int(x::Float32) = Int(10 * x)
-        @test estimate_magitude(f32, 0f0) === 0.1f0
-        @test estimate_magitude(f32, 1f0) === 1f0
-        # In this case, the `Int` is converted with `float`, so we expect a `Float64`.
-        @test estimate_magitude(f32_int, 0f0) === 1.0
-        @test estimate_magitude(f32_int, 1f0) === 10.0
-    end
-
-    @testset "estimate_roundoff_error" begin
-        # `Float64`s:
-        @test estimate_roundoff_error(identity, 1.0) == eps(1.0)
-        #   Pertubation from `estimate_magitude`:
-        @test estimate_roundoff_error(identity, 0.0) == eps(0.1)
-
-        # `Float32`s:
-        @test estimate_roundoff_error(identity, 1f0) == eps(1f0)
-        #   Pertubation from `estimate_magitude`:
-        @test estimate_roundoff_error(identity, 0.0f0) == eps(0.1f0)
+    @testset "Correctness" begin
+        # The different approaches to approximating the gradient to try.
+        methods = [forward_fdm, backward_fdm, central_fdm]
+
+        # The different floating point types to try and the associated required relative
+        # tolerance.
+        types = [Float32, Float64]
+
+        # The different functions to evaluate (`.f`), their first derivative at 1 (`.d1`),
+        # and second derivative at 1 (`.d2`).
+        foos = [
+            (f=sin, d1=cos(1), d2=-sin(1)),
+            (f=exp, d1=exp(1), d2=exp(1)),
+            (f=abs2, d1=2, d2=2),
+            (f=x -> sqrt(max(x + 1,0)), d1=0.5 / sqrt(2), d2=-0.25 / 2^(3/2)),
```

Comment:
Would you consider adding the function suggested in #124 here, or does it continue to be tested elsewhere?

Reply:
Hmm, I like to think of this test set as one that tests for correctness rather than expected accuracy. We can similarly aggregate a testset that tests for accuracy. I think that would be good.